### PR TITLE
Silence grpc connection log noise.

### PR DESCRIPTION
**Pull request by retroflexer**
A hack to eliminate the superfluous grpc connection log spam from etcd client. Warnings and errors are still emitted.
kubernetes/kubernetes#80741
kubernetes/kubernetes#84061
https://github.com/openshift/cluster-etcd-operator/pull/368
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
To eliminate the superfluous grpc connection log spam from etcd client. 
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by retroflexer** [comment]
I am looking the old thread:
https://github.com/kubernetes/kubernetes/pull/84061/files#r336218221
I will push another change with klog writers. 
Also need to update the vendor.

**Comment by retroflexer** [comment]
I signed it.

File: staging/src/k8s.io/apiserver/pkg/storage/etcd3/logger.go [hunk 1]
```diff
@@ -32,15 +32,15 @@ type klogWrapper struct{}
...
 }
 
 func (klogWrapper) Infoln(args ...interface{}) {
-	klog.InfoDepth(klogWrapperDepth, fmt.Sprintln(args...))
+	//Discard all Info logging. Do nothing.
 }
 
 func (klogWrapper) Infof(format string, args ...interface{}) {
-	klog.InfoDepth(klogWrapperDepth, fmt.Sprintf(format, args...))
+	//Discard all Info logging. Do nothing.
```
Comment (sttts) [inline staging/src/k8s.io/apiserver/pkg/storage/etcd3/logger.go:43]: Depending on verbosity level we still want these. Maybe at 9 ? Don't know how we map that elsewhere.

Comment (sttts) [reply staging/src/k8s.io/apiserver/pkg/storage/etcd3/logger.go:43]: Or maybe 4 or 6.

**Comment by sttts** [comment]
cc @hexfusion

File: staging/src/k8s.io/apiserver/pkg/storage/etcd3/logger.go (hunk 1 above)
Comment (ehashman) [reply staging/src/k8s.io/apiserver/pkg/storage/etcd3/logger.go:43]: Agree, seems needless to totally throw this away... can we not bump it to a more verbose log level?

**Comment by aojea** [comment]
this is super hacky :upside_down_face:

**Comment by liggitt** [comment]
I'll defer to some of the etcd folks on whether this is a good approach - cc @ptabor / @mm4tt

**Comment by retroflexer** [comment]
> this is super hacky upside_down_face

Why do you think it is hacky? You may notice that the logwrapper was already there before this PR, and I just used the existing log wrapper to silence the grpc log noise.

**Comment by ptabor** [comment]
I think that etcd client 3.5 have a lot of improvement in that area: 
  - Since https://github.com/etcd-io/etcd/pull/12671  etcd client does not use custom balancer, just the standard one, so the logging should be the same as all the other services. 
  - Control using `ETCD_CLIENT_DEBUG` env variable (https://github.com/etcd-io/etcd/pull/12786)

**Comment by aojea** [comment]
> Why do you think it is hacky? You may notice that the logwrapper was already there before this PR, and I just used the existing log wrapper to silence the grpc log noise.

well, it is in the description :)

> A hack to eliminate the superfluous grpc connection log spam from etcd client. Warnings and errors are still emitted.

I have some questions, because I know and agree that the grpc logs are to verbose, but:
- Is there consensus that ALL the info grpc level is noise? the grpc-go people doesn't seem to have the same opinion

https://github.com/grpc/grpc-go/pull/2943#issuecomment-516881207
https://github.com/grpc/grpc-go/blob/master/Documentation/log_levels.md

- what if there is something that "is not noise" later? 
debug 6 is too verbose, there is no way to enable that in production just to see a grpc message

**Comment by retroflexer** [comment]
closing as the replicated PR 102896 got merged.

/close

**Pull request closed without merging**